integration-cli/docker_cli_build_test.go

```diff
@@ -7218,3 +7218,39 @@ func (s *DockerSuite) TestBuildLabelsOverride(c *check.C) {
 	}
 
 }
+
+// Shell test to confirm config gets updated correctly
+func (s *DockerSuite) TestBuildShellUpdatesConfig(c *check.C) {
+	name := "testbuildshellupdatesconfig"
+
+	expected := `["foo","-bar","#(nop) ","SHELL foo -bar"]`
+	_, err := buildImage(name,
+		`FROM `+minimalBaseImage()+`
+        SHELL foo -bar`,
+		true)
+	if err != nil {
+		c.Fatal(err)
+	}
+	res := inspectFieldJSON(c, name, "ContainerConfig.Cmd")
+	if res != expected {
+		c.Fatalf("%s, expected %s", res, expected)
+	}
+}
```

Comment:
I'd like to see a testcases that:
- uses default shell
- sets SHELL to something
- verifies that RUN uses new SHELL
- sets SHELL to something-else
- verifies that RUN uses "something-else"
- uses CMD and ENTRYPOINT
- verifies that resulting image uses "something-else" when run

also:
- use CMD and ENTRYPOINT
- then set SHELL
- verify the correct shell is used in resulting image. Which is it? default or SHELL value? I'm guessing the default - but I'm not sure.  We need to make the docs state this.

and:
- do two builds where build 1 sets SHELL to something
- then build 2, which uses the image from build 1, verifies that it picks up the new SHELL from build 1 for RUN, CMD and ENTRYPOINT when those cmds are used in build 2